tests/trainer/test_dataloaders.py

```diff
@@ -1075,44 +1069,108 @@ def test_dataloaders_load_only_once_no_sanity_check(tmpdir):
     tracker.attach_mock(model.test_dataloader, "test_dataloader")
 
     trainer.fit(model)
+    trainer.test(model)
+
+    # Verify the sequence
+    expected_sequence = [call.val_dataloader(), call.train_dataloader()]  # Sanity check first
+    if n == 1:
+        expected_sequence += [call.train_dataloader(), call.val_dataloader()] * 4
+    elif n == 2:
+        expected_sequence += [call.train_dataloader(), call.val_dataloader()] * 2
+    expected_sequence += [call.test_dataloader()]
 
-    # verify the sequence
-    expected_sequence = [call.train_dataloader(), call.val_dataloader()]
     assert tracker.mock_calls == expected_sequence
 
+    # Verify epoch of reloads
+    if n == 1:
+        assert train_reload_epochs == [0, 1, 2, 3, 4]
+        assert val_reload_epochs == [0, 1, 2, 3, 4]
+    elif n == 2:
+        assert train_reload_epochs == [0, 2, 4]
+        assert val_reload_epochs == [0, 2, 4]
 
-@pytest.mark.parametrize("n", [1, 2])
-def test_dataloaders_load_every_n_epochs(tmpdir, n):
-    model = BoringModel()
+
+@pytest.mark.parametrize("n", [3, 5])
+def test_dataloaders_load_every_n_epochs_infrequent_val(tmpdir, n):
```

Comment:
can you add a small docstring to these tests explaining in one sentence what is being tested?

Reply:
Added!